include/clang/AST/Expr.h

```diff
@@ -2798,7 +2798,9 @@ class BoundsExpr : public Expr {
 /// classes).
 class CastExpr : public Expr {
 private:
-  enum { OP, BOUNDS, END_EXPR = 2 };
+  // BOUNDS - enum value for bounds of cast expression
+  // SUBBOUNDS - enum value for bounds of subexpression
+  enum { OP, BOUNDS, SUBBOUNDS, END_EXPR = 3 };
```

Comment:
I desinged as follows:
In bounds inferference, count/byte_count bounds information is expanded into range bounds information. Also, In code generation, it requires range bounds expression.
So I have overwritten bounds expression in bounds inference
In code generation, it requires also source bounds expression to compare bounds expression of cast operation. Since I do not have any idea to pass source bounds expression in code generation, I have set source bounds expression when bounds inference
I think that it is correct to expand range bounds expression of bounds cast expression in semantic analysis phase because code generation does not handle this
How about your idea?